rules/no-specific-file-extensions.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const path = require('path');
+
+const DEFAULTS = {
+  extensions: ['.jsx']
+};
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Restrict specific file extensions',
+      category: 'Best Practices',
+      recommended: false
+    },
+
+    schema: [{
+      type: 'object',
+      properties: {
+        extensions: {
+          type: 'array',
+          items: {
+            type: 'string'
+          }
+        }
+      },
+      additionalProperties: false
+    }]
+  },
+
+  create: function(context) {
+    function getExtensionsConfig() {
+      return context.options[0] && context.options[0].extensions || DEFAULTS.extensions;
+    }
+
+    return {
+      Program: function(node) {
+        const filename = context.getFilename();
+
+        if (filename === '<text>') {
+          return;
+        }
+
+        const extension = path.extname(filename);
+        const allowedExtensions = getExtensionsConfig();
+        const isRestrictedExtension = allowedExtensions.includes(extension);
+
+        if (!isRestrictedExtension) {
+          return;
+        }
+
+        context.report({
+          node: node,
+          message: `File extension '${extension}' not allowed.`
```

Comment:
Maybe adding the allowed extensions would help users to know how to fix the problem? Something like `Please use one of these extensions: ${allowedExtensions.join(', ')}`.

Reply:
This code is quite odd. `allowedExtensions` should be `notAllowedExtensions`, since the config specify restricted ones. So we don't have the ones that are allowed.

Reply:
I see. Do you want to change it?

Reply:
Yes, looking better at it I have to review this PR before we merge it. Let's put it on hold. 

Maybe we can specify pull reminders to not notify PRs that have the label `status: on hold`.